File: backend/app/domain/schemas/applications/commission.py
from typing import Any
from datetime import datetime

from pydantic import BaseModel, validator, Field

from app.domain.schemas.application import ApplicationResponse


class CommissionBase(BaseModel):
    country: str
    state: str | None
    city: str | None
    start_date: datetime
    end_date: datetime
    lenguage: str | None
    justification: str = Field(max_length=500, min_length=5)
    documents: list[Any] | None

class CommissionCreate(CommissionBase):
    application_sub_type_id: int = Field(gt=0, lt=100)


class CommissionUpdate(CommissionBase):
    application_sub_type_id: int


class Compliment(BaseModel):
    documents: list[Any]
    emails: list[str]
    observation: str = Field(max_length=300)


class commissionInDB(CommissionBase):
    resolution: str | None
    compliment: Compliment | None


class CommissionResponse(ApplicationResponse):
    commission: commissionInDB


class CommissionDocument(commissionInDB):
    @validator("start_date", 'end_date')
    def stringdate(cls, v, values, **kwargs):
        return v.strftime("%A %d de %B del %Y")

File: client/src/app/core/interfaces/applications/commission.ts
import { Application } from "@interfaces/application";
import { file_path } from "@interfaces/documents";

export interface CommissionCreate {
    application_sub_type_id: number;
    country: string;
    state?: string;
    city?: string;
    start_date: Date;
    end_date: Date;
    reason: string;
    lenguage: string;
    justification: string;
    documents?: file_path[];
}


export interface CommissionInDB extends CommissionCreate {
    id: string;
    resolution?: string;
    compliment?: ComplimentCreate;
}


export interface CommissionResponse extends Application {
    commission: CommissionInDB;
}


export interface ComplimentCreate {
    documents: file_path[];
    emails: string[];
    observation: string;
}
